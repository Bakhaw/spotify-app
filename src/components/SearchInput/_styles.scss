.SearchInput {
  display: flex;
  align-items: center;
  background: #fff;
  border: 3px solid #ebedf4;
  border-radius: 50px;
  padding: 10px 15px;

  img {
    height: 25px;
    width: 25px;
  }

  input {
    border-radius: 50px;
    border: none;
    outline: none;
    color: #3d445b;
    font-size: 14px;
    padding-left: 10px;
    height: 100%;
    width: 100%;
  }
}
