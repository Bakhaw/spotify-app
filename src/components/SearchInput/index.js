import React, { useState } from 'react';

import searchIcon from '../../icons/icon-search.svg';

function SearchInput({ searchArtist }) {
  const [inputValue, setInputValue] = useState('');

  function onChange(e) {
    setInputValue(e.target.value);
  }

  function onKeyPress(e) {
    if (e.key === 'Enter') {
      e.target.blur();
      e.preventDefault();
      searchArtist(inputValue);
    }
  }

  return (
    <div className='SearchInput'>
      <img alt='Search' src={searchIcon} />

      <input onChange={onChange} onKeyPress={onKeyPress} />
    </div>
  );
}

export default SearchInput;
