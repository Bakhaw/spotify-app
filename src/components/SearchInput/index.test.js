import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import SearchInput from '.';

test('components/SearchInput/', () => {
  const { container } = render(<SearchInput />);
  expect(container.firstChild).toMatchSnapshot();
});
