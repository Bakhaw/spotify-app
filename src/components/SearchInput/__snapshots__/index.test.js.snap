// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Input/ 1`] = `
<input
  class="Input"
/>
`;
